### PR TITLE
[16][MIG] auth_api_key_server_env #455

### PR DESCRIPTION
**Pull request by florian-dacosta**
I open a new PR for this one, as last one was closed because stale.
But I believe this module is still needed, maybe a small review for past contributors/reviewers could help moving forward
@simahawk @qgroulard @yankinmax @SilvioC2C

### CODE REVIEW

**Review by sebalix — APPROVED** [review]

File: auth_api_key_server_env/models/auth_api_key.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+from odoo import models
+
+
+class AuthApiKey(models.Model):
+
+    _name = "auth.api.key"
+    _inherit = ["auth.api.key", "server.env.mixin"]
+
+    def _server_env_section_name(self):
```
Comment (simahawk) [inline auth_api_key_server_env/models/auth_api_key.py:15]: I would drop this and use `server.env.techname.mixin` to not rely on the name which can be used for something more meaningful and not screwed by occasional editing. WDYT?

Comment (florian-dacosta) [reply auth_api_key_server_env/models/auth_api_key.py:15]: It makes sense, I made the change and added a migration script.

File: auth_api_key_server_env/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Configure api keys via server env.
+
+This can be very useful to avoid mixing your keys between your various
+environments when restoring databases. All you have to do is to add a new
+section to your configuration file according to the following convention:
+
+.. code-block:: ini
+
+    [api_key_<Record Name>]
```
Comment (simahawk) [inline auth_api_key_server_env/readme/DESCRIPTION.rst]:
```suggestion
    [api_key_<record.tech_name>]
```

File: auth_api_key_server_env/readme/CONTRIBUTORS.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+* Simone Orsi <simone.orsi@camptocamp.com>
```
Comment (simahawk) [inline auth_api_key_server_env/readme/CONTRIBUTORS.rst:1]: You can list yourself here :wink:

File: auth_api_key_server_env/tests/test_auth_api_key.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class TestAuthApiKey(TransactionCase):
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super().setUpClass(*args, **kwargs)
+        cls.AuthApiKey = cls.env["auth.api.key"]
+        cls.demo_user = cls.env.ref("base.user_demo")
+        cls.api_key_from_env = cls.AuthApiKey.create(
+            {"name": "from_env", "key": "dummy", "user_id": cls.demo_user.id}
```
Comment (simahawk) [inline auth_api_key_server_env/tests/test_auth_api_key.py]: can you use tech_name here w/ a different value from name?

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by florian-dacosta** [comment]
@simahawk I have fixed pre-commit and the test + and to your comments, thanks

File: auth_api_key_server_env/models/auth_api_key.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        """Name of the section in the configuration files
+        We override the default implementation to keep the compatibility
+        with the previous implementation of auth_api_key. The section name
+        into the configuration file must be formatted as
+            'api_key_{name}'
+        """
+        self.ensure_one()
+        return "api_key_{}".format(self.tech_name)
```
Comment (simahawk) [inline auth_api_key_server_env/models/auth_api_key.py]:
```suggestion
        return "api_key_{}".format(self._server_env_section_name_field)
```

**Review by simahawk — APPROVED** [review]
Can you rewrite the commit to `auth_api_key_server_env: use tech_name`?
Pre-approving

**Comment by simahawk** [comment]
Food for thoughts: we might want to refactor this and the main module in favor of core `res.users.apikeys` model.

/ocabot merge nobump

**Pull request merged**